app/herdbook.py

```diff
@@ -9,16 +9,23 @@
 import sys
 import uuid
 import time
+import json
 from flask import (
     Flask,
     jsonify,
     request,
     session,
+    redirect,
+    url_for
 )
-from flask_caching import Cache
+
+from werkzeug.urls import url_parse
 import utils.database as db
 import utils.data_access as da
 import utils.inbreeding as ibc
+import logging
+from flask_caching import Cache
+from flask_login import login_required, login_user, logout_user, current_user, LoginManager
```

Comment:
I don't mind using the Flask login system, but I'm not a big fan of using different systems on different endpoints. I seem to remember that the flask login system didn't really play that well with asynchronous requests, but if you can get it to work - please change all the endpoints so that the app remains consistent! :)

Reply:
I did not touch your login system, just added what I needed to use the login_required attribute on the python methods. This does not change the react form or any logic behind. But I would like to be able to implement the return to an url after login, for example. For that, some integration with react needs to be done, since the return url is sent to the login page

Reply:
Absolutely! But if we use decorators for one endpoint we should use them for all endpoints that require login so that the system remains consistent. Adding a login-redirector sounds like a great plan though! :)

Reply:
Sure. I just did not wanted to do that without your approval

Reply:
done